odoo/fields.py

```diff
@@ -1294,6 +1294,80 @@ def convert_to_export(self, value, record):
     compare = staticmethod(float_compare)
 
 
+class Uom(Field):
+    """ The decimal precision is taken from the attribute
+
+    :param uom_field: name of the field holding the uom this uom
+                           field is expressed in (default: `uom_id`)
+    """
+    type = 'uom'
+    column_type = ('numeric', 'numeric')
+    column_cast_from = ('float8',)
+    _slots = {
+        'uom_field': None,
+        'group_operator': 'sum',
+    }
+
+    def __init__(self, string=Default, uom_field=Default, **kwargs):
+        super(Uom, self).__init__(string=string, uom_field=uom_field, **kwargs)
+
+    _description_uom_field = property(attrgetter('uom_field'))
+
+    def _setup_uom_field(self, model):
+        if not self.uom_field:
+            # pick a default, trying in order: 'uom_id', 'x_uom_id'
+            if 'uom_id' in model._fields:
+                self.uom_field = 'uom_id'
+            elif 'x_uom_id' in model._fields:
+                self.uom_field = 'x_uom_id'
+        assert self.uom_field in model._fields, \
+            "Field %s with unknown uom_id %r" % (self, self.uom_field)
+
+    def _setup_regular_full(self, model):
+        super(Uom, self)._setup_regular_full(model)
+        self._setup_uom_field(model)
+
+    def _setup_related_full(self, model):
+        super(Uom, self)._setup_related_full(model)
+        if self.inherited:
+            self.uom_field = self.related_field.uom_field
+        self._setup_uom_field(model)
+
+    def write(self, records, value):
+        # Manage uom for multi records
+        for record in records:
+            super(Uom, self).write(record, value)
```

Comment:
Why is this `write` override needed ? Is there a reason why the field can't be written in batch ?
I don't see any equivalent override in the Monetary field, which is the kind of field nearest to the Uom field (but it was maybe removed in Monetary fields, in which case we can maybe also remove it here).

Reply:
Indeed, this override is nonsense.